### PR TITLE
Improve resiliency of CertificateValidationRemoteServer_EndToEnd_Ok test

### LINKED ISSUE #2350: CertificateValidationRemoteServer_EndToEnd_Ok test failing on all PRs on Windows
cc: @wfurt, @davidsh

### PR DESCRIPTION
**Pull request by wfurt**
This test connects to Azure endpoint and depends on external services.
Once in a while, we seek a spike of failures. This is not specific to Windows 7 (https://github.com/dotnet/corefx/issues/42339) but more so luck and overall network condition. 

Since this test is trying to verify certificate validation, I made modifications to ignore connect failures as well as cases when we get reset from the remote peer. That is most likely caused by the network infrastructure and not caused by the product. If we mess up handshake we would most likely get AuthenticationException and the test would still fail.

fixes  #2350
contributes to https://github.com/dotnet/corefx/issues/42339

### CODE REVIEW

**Review by stephentoub — APPROVED** [review]
Seems reasonable. Thanks.

File: src/libraries/System.Net.Security/tests/FunctionalTests/CertificateValidationRemoteServer.cs [hunk 1]
```diff
@@ -15,23 +17,45 @@ namespace System.Net.Security.Tests
...
-                await client.ConnectAsync(Configuration.Security.TlsServer.IdnHost, Configuration.Security.TlsServer.Port);
+                try
+                {
+                    await client.ConnectAsync(Configuration.Security.TlsServer.IdnHost, Configuration.Security.TlsServer.Port);
+                }
+                catch (Exception ex)
+                {
+                    // if we cannot connect, skip the test instead of failing.
+                    // This test is not trying to test networking.
+                    new SkipTestException($"Unable to connect to '{Configuration.Security.TlsServer.IdnHost}': {ex.Message}");
```
Comment (davidsh) [inline src/libraries/System.Net.Security/tests/FunctionalTests/CertificateValidationRemoteServer.cs]: I think this should be `throw new SkipTestException()'.  Doing just `new SkipTestException()` won't do anything.

File: src/libraries/System.Net.Security/tests/FunctionalTests/CertificateValidationRemoteServer.cs [hunk 2]
```diff
@@ -15,23 +17,45 @@ namespace System.Net.Security.Tests
...
+                        {
+                            sslStream.AuthenticateAsClient(Configuration.Security.TlsServer.IdnHost);
+                        }
+                    }
+                    catch (IOException ex) when (ex.InnerException is SocketException &&
+                      ((SocketException)ex.InnerException).SocketErrorCode == SocketError.ConnectionReset)
+                    {
+                        // Since we try to verify certificate validation, ignore IO errors
+                        // caused most likely by environmental failures.
+                        new SkipTestException($"Unable to connect to '{Configuration.Security.TlsServer.IdnHost}': {ex.InnerException.Message}");
```
Comment (davidsh) [inline src/libraries/System.Net.Security/tests/FunctionalTests/CertificateValidationRemoteServer.cs]: I think this should be `throw new SkipTestException()'.  Doing just `new SkipTestException()` won't do anything.

File: src/libraries/System.Net.Security/tests/FunctionalTests/CertificateValidationRemoteServer.cs (hunk 1 above)
Comment (wfurt) [reply src/libraries/System.Net.Security/tests/FunctionalTests/CertificateValidationRemoteServer.cs]: good catch, fixed.

**Pull request merged**